pandas/core/internals/managers.py

```diff
@@ -1941,6 +1941,24 @@ def _compare_or_regex_search(a, b, regex=False):
     -------
     mask : array_like of bool
     """
+
+    def _check(result, a, b):
```

Comment:
can you give this a more informative name, maybe _check_comparision_types

and type the input args as much as possible